### PR TITLE
Added templ generate to workflow

### PR DESCRIPTION
**Pull request by brijesh-amin**
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

## Problem/Feature

Added templ generate to workflow #185 

## Description of Changes: 

- Added templ generate to go-test
- Added templ generate to goreleaser

## Checklist

- [x] I have self-reviewed the changes being requested
- [x] I have updated the documentation (check issue ticket #185 )

### CODE REVIEW

File: cmd/template/advanced/files/workflow/github/github_action_releaser_config.yml.tmpl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+version: 2
```
Comment (Ujstor) [inline cmd/template/advanced/files/workflow/github/github_action_releaser_config.yml.tmpl:1]: Does version 2 refer to goreleaser v2?

Comment (brijesh-amin) [reply cmd/template/advanced/files/workflow/github/github_action_releaser_config.yml.tmpl:1]: Yes, without that, I noticed some errors

Comment (Ujstor) [reply cmd/template/advanced/files/workflow/github/github_action_releaser_config.yml.tmpl:1]: In that case, goreleaser/goreleaser-action@v6 is needed in the release workflow. I didn't test v2, please check if everything is working as expected with the current .goreleaser.yml configuration. I thinkl the easiest way is to create a test repo on GitHub and push a tag.

Comment (Ujstor) [reply cmd/template/advanced/files/workflow/github/github_action_releaser_config.yml.tmpl:1]: I tested releaser in the live repository, and it works as expected.

File: cmd/template/advanced/files/workflow/github/github_action_goreleaser.yml.tmpl [hunk 2]
```diff
@@ -16,14 +16,21 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: '1.21.1'
-      -
-        name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v5.0.0
+          go-version: '1.22.x'
+        {{if or ( .AdvancedOptions.htmx ) ( .AdvancedOptions.tailwind )}}
```
Comment (Ujstor) [inline cmd/template/advanced/files/workflow/github/github_action_goreleaser.yml.tmpl:20]: Indent all template statements to the beginning of the line. In YAML, curent indentation causes unwanted spaces to be inserted

File: cmd/template/advanced/files/workflow/github/github_action_goreleaser.yml.tmpl [hunk 3]
```diff
@@ -16,14 +16,21 @@ jobs:
...
+      - name: Install templ
+        shell: bash
+        run: go install github.com/a-h/templ/cmd/templ@latest
+      - name: Run templ generate
+        shell: bash
+        run: templ generate -path .
+        {{end}}
```
Comment (Ujstor) [inline cmd/template/advanced/files/workflow/github/github_action_goreleaser.yml.tmpl:27]: Same as previously

File: cmd/template/advanced/files/workflow/github/github_action_gotest.yml.tmpl [hunk 4]
```diff
@@ -10,8 +10,16 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: '1.21.x'
+          go-version: '1.22.x'
+        {{if or ( .AdvancedOptions.htmx ) ( .AdvancedOptions.tailwind )}}
```
Comment (Ujstor) [inline cmd/template/advanced/files/workflow/github/github_action_gotest.yml.tmpl:14]: Do same thing for this file.

**Review by Ujstor — CHANGES_REQUESTED** [review]

**Comment by Ujstor** [comment]
Previous push didn't solve the issue. Indent all template statements to the beginning of the line. Use indentation as below:

```bash
name: Go-test
on: [push, pull_request]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4
      - name: Setup Go
        uses: actions/setup-go@v4
        with:
          go-version: '1.22.x'
{{if or ( .AdvancedOptions.htmx ) ( .AdvancedOptions.tailwind )}}
      - name: Install templ
        shell: bash
        run: go install github.com/a-h/templ/cmd/templ@latest
      - name: Run templ generate
        shell: bash
        run: templ generate -path .
{{end}}
      - name: Build
        run: go build -v ./...
      - name: Test with the Go CLI
        run: go test ./... 
```

**Comment by brijesh-amin** [comment]
@Ujstor I have modified accordingly. Thanks.

**Comment by Ujstor** [comment]
> @Ujstor I have modified accordingly. Thanks.

You're welcome, we are ready for merge.

**Review by Ujstor — APPROVED** [review]

**Pull request merged**